File: TokenManager/src/main/java/com/hSenid/demo/models/Patient.java
package com.hSenid.demo.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.Transient;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.LocalDate;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document (collection = "patients")
public class Patient implements Serializable {

    @Transient
    public static String SEQUENCE_NAME = "user_sequence";

    @Id
    private int id;
    private String firstName;
    private String email;
    private String lastName;
    private Gender gender;
    private LocalDate dob;
    private Long contactNum;
}

File: angular-16-jwt-auth-master/src/app/models/patient.ts
import { Gender } from "./gender.enum";

export interface Patient{
    id?: number;
    firstName?: string;
    lastName?: string;
    gender?: Gender;
    dob?: Date;
    contactNum?: number;
    username?: string;
    email?: string;
    password?: string;
    roles?: Array<string>;
}

export class PatientUpdateRequest implements Patient {
    constructor(
        public id?: number,
      public firstName?: string,
      public lastName?: string,
      public gender?: Gender,
      public dob?: Date,
      public contactNum?: number,
      public email?: string,
      public username?: string
    ) {}
  }